shorebird_code_push/lib/src/shorebird_updater.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:shorebird_code_push/src/shorebird_updater_io.dart'
+    if (dart.library.js_interop) './shorebird_updater_web.dart';
+import 'package:shorebird_code_push/src/updater.dart';
+
+/// The reason a call to [ShorebirdUpdater.update] failed.
+enum UpdateFailureReason {
+  /// No update is available.
+  noUpdate,
+
+  /// The update failed because the patch failed to install.
+  installFailed,
```

Comment:
Nit, I would reorder downlaod before install, since it happens before install, but 🤷